lib/evmone/eof.cpp

```diff
@@ -61,48 +61,33 @@ std::variant<EOFSectionHeaders, EOFValidationError> validate_eof_headers(bytes_v
     auto it = container.begin() + std::size(MAGIC) + 1;  // MAGIC + VERSION
     // TODO: Since all sections are mandatory and they have to be ordered (Types, Code+, Data)
     // TODO: this fragment of code can be much simpler. Rewriting needed.
+    uint8_t expected_section_id = TYPE_SECTION;
     while (it != container_end && state != State::terminated)
     {
         switch (state)
         {
         case State::section_id:
         {
             section_id = *it++;
+
+            if (section_id != expected_section_id)
+                return EOFValidationError::expected_section_missing;
```

Comment:
Here you can select error code by `expected_section_id` so we preserve more detailed error information. If you want to be fancy, you can do something like
`return errc::type_section_missing + expected_section_id`.